controllers/cluster_controller_phases.go

```diff
@@ -64,6 +64,24 @@ func (r *ClusterReconciler) reconcilePhase(_ context.Context, cluster *clusterv1
 func (r *ClusterReconciler) reconcileExternal(ctx context.Context, cluster *clusterv1.Cluster, ref *corev1.ObjectReference) (*unstructured.Unstructured, error) {
 	logger := r.Log.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
 
+	// Add watcher for external object, if there isn't one already.
+	_, loaded := r.externalWatchers.LoadOrStore(ref.GroupVersionKind().String(), struct{}{})
+	if !loaded && r.controller != nil {
+		logger.Info("Adding watcher on external object", "gvk", ref.GroupVersionKind())
+		obj := new(unstructured.Unstructured)
```

Comment:
My thought is that we could eventually eliminate having to re-queue when retrieving the external resource fails, by enabling the watch first.

Reply:
I had originally tried to do a quick refactor of that, but it's a bit tricky because without throwing the requeueError it attempts to continue reconciling instead of short circuiting and not requeuing the request.

Reply:
I can revert this change if it makes sense, since it'll require additional changes to see any real benefit.

Reply:
The reason the Watch was only called after the initial `Get` was to make sure that the CRDs and the object existed before the call, and it would save you from setting the APIVersion, Kind, etc

Reply:
Yeah, I get that, however having the watch first means that we don't have to do a requeue if the referenced resource is created after the resource that is referencing it.

Reply:
Can we double-check the implications of adding a watch to a CRD that doesn't exist? If it errors out and it doesn't loop forever, that's probably ok

Reply:
I'll go ahead and back out this change, since we are still requeueing anyway...  We can revisit it at a later time.